Project-Lily/Project-Lily/DogListViewModel.swift

```diff
@@ -8,15 +8,20 @@
 
 import Foundation
 
-//class ViewModel {
-//    var dogs = [Dog]()
-//
-//    func getDogs() {
-//        let gotDogsCompletion: ([Dog]) -> Void = { newDogs in
-//                self.dogs = newDogs
-//                // delegate?.didFetchDogs()
-//        }
-//
-//        NetworkManager().createDogURLSession(completion: gotDogsCompletion)
-//    }
-//}
+protocol DogListViewModelDelegate: class {
+    func didFetchDogs()
+}
+
+class DogListViewModel {
+    var dogs = [Dog]()
+   weak var delegate: DogListViewModelDelegate?
+
+    func getDogs() {
+        let gotDogsCompletion: ([Dog]) -> Void = { newDogs in
+                self.dogs = newDogs
+                print(self.dogs)
+                self.delegate?.didFetchDogs()
+        }
+        NetworkManager().fetchDogs(completion: gotDogsCompletion)
```

Comment:
🍰 It *might* be good to incorporate the closure within the `fetchDogs` call itself, instead of passing it in as a variable. Only mentioning this because it is typical in JJs to do so (and I usually see other iOS developers do the same). Might make for good practice.

Reply:
- [ ] Incorporate closure within method call